websocket.go

```diff
@@ -135,25 +155,31 @@ func (c *Connection) handleResponse() {
 			}
 			log.Println(r)
 		}
+		c.clientMap.lock.RUnlock()
+		c.subscriberMap.lock.RUnlock()
 	}
 }
 
 // Allow clients to subscribe to messages intended for them
 func (c *Connection) Subscribe(eventType string, elementId string) <-chan Response {
-	if c.subscribers == nil {
-		c.subscribers = make(map[string]map[string]chan Response)
+	if c.subscriberMap.subscribers == nil {
+		c.subscriberMap.subscribers = make(map[string]map[string]chan Response)
```

Comment:
Probably want to lock this write operation?

Reply:
I figure if the map doesn't exist yet, nothing will be able to read from it anyway?

Reply:
Someone else might do the same operation at the same time, or just after someone has initialised the map and a value? eg lose that value?

Reply:
True, good point. Have updated to catch that.